src/librustc_mir/transform/qualify_min_const_fn.rs

```diff
@@ -92,7 +93,15 @@ pub fn is_min_const_fn(tcx: TyCtxt<'tcx>, def_id: DefId, body: &'a Body<'tcx>) -
 }
 
 fn check_ty(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>, span: Span, fn_def_id: DefId) -> McfResult {
-    for ty in ty.walk() {
+    for arg in ty.walk() {
+        let ty = match arg.unpack() {
```

Comment:
I guess we don't have a convenient method to do `for arg in ty.walk().filter_map(|arg| arg.ty())` or something?

I'm guessing you're going to tell me that it's worth writing the match anyway, so that we can document why we only care about types (which I would believe)

Reply:
> I'm guessing you're going to tell me that it's worth writing the match anyway, so that we can document why we only care about types (which I would believe)

Yeah, and it actually found a bunch of bugs where consts weren't handled.